src/codegen/c.zig

```diff
@@ -3149,9 +3201,18 @@ fn airStructFieldVal(f: *Function, inst: Air.Inst.Index) !CValue {
     const writer = f.object.writer();
     const struct_byval = try f.resolveInst(extra.struct_operand);
     const struct_ty = f.air.typeOf(extra.struct_operand);
+    var buf = std.ArrayList(u8).init(f.object.dg.gpa);
+    defer buf.deinit();
     const field_name = switch (struct_ty.tag()) {
         .@"struct" => struct_ty.structFields().keys()[extra.field_index],
         .@"union", .union_tagged => struct_ty.unionFields().keys()[extra.field_index],
+        .tuple => blk: {
+            const tuple = struct_ty.tupleFields();
+            if (tuple.values[extra.field_index].tag() != .unreachable_value) return CValue.none;
```

Comment:
I might be mistaken, but I don't think the backend should ever be asked to produce the field value of a comptime field.

Is it possible to assert here that the field is not comptime instead?

Reply:
I don't think that's necessary, see:

https://github.com/ziglang/zig/blob/f3f5a5d05b7056aceb408b701613242d053019ab/src/type.zig#L5528-L5529

`tuple.values[i] == .unreachable_value` should be sufficient to tell that the value is runtime-known.

Reply:
Yeah - I meant to recommend you assert the tag is `.unreachable_value` (rather than returning `CValue.none`)

However, I just checked that `CValue.none` already asserts if we try to render it anywhere, so this is fine as-is :+1: